### PR TITLE
Don't use nested Promises

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

* Introduces the `promise/no-nesting` rule in our ESLint config (Commented, as this is a multiple PR effort)
* Converts a bunch of nested promises to chained `then`.

**Issues**

### CODE REVIEW

File: src/components/collectionEditor/collectionEditor.js [hunk 1]
```diff
@@ -67,23 +67,20 @@ import 'flexStyles';
...
             Ids: dlg.querySelector('.fldSelectedItemIds').value || ''
         });
 
-        apiClient.ajax({
-            type: 'POST',
-            url: url
-
-        }).then(() => {
+        apiClient.ajax({ type: 'POST', url: url })
+        .then(() => {
```
Comment (dkanada) [inline src/components/collectionEditor/collectionEditor.js:74]:
```suggestion
        apiClient.ajax({ type: 'POST', url: url }).then(() => {
```

Comment (dkanada) [reply src/components/collectionEditor/collectionEditor.js:74]: You don't like this method?

Comment (MrTimscampi) [reply src/components/collectionEditor/collectionEditor.js:74]: I find putting them on the next line more readable. It's easier to see at a glance where the different blocks are.

It's also what Prettier uses, so if we ever decide to use that (there has been talks about it before), it's one less change (We're already not very far from the style it uses).

Comment (YouKnowBlom) [reply src/components/collectionEditor/collectionEditor.js:74]: I agree, newline looks much cleaner when chaining

**Pull request closed without merging**